tests/users/test_views.py

```diff
@@ -0,0 +1,136 @@
+    assert list(resp.context['form'].errors.values()) == [[error_messages['name_error']]]
+    assert User.objects.count() == 1
+
+
+@pytest.mark.django_db
+def test_register_with_name_with_numbers(client: Client, user, error_messages):
+    resp = client.post(reverse('register'), data={'first_name': "T3st", 'last_name': 'User', 'email': "test_@test.com",
+                                                  'password': "test123?", 'agree_to_terms': True})
+    assert list(resp.context['form'].errors.values()) == [[error_messages['name_error']]]
+    assert User.objects.count() == 1
+
+
+@pytest.mark.django_db
+def test_register_with_existing_email(client: Client, user, error_messages):
+    resp = client.post(reverse('register'), data={'first_name': "Test", 'last_name': 'User', 'email': "test@test.com",
+                                                  'password': "test123?", 'agree_to_terms': True})
+    assert list(resp.context['form'].errors.values()) == [[error_messages['email_error']]]
+    assert User.objects.count() == 1
+
+
+@pytest.mark.django_db
+def test_register_with_short_password(client: Client, user, error_messages):
+    resp = client.post(reverse('register'), data={'first_name': "Test", 'last_name': 'User', 'email': "test_@test.com",
+                                                  'password': "test1?", 'agree_to_terms': True})
+    assert list(resp.context['form'].errors.values()) == [[error_messages['password_error']]]
+    assert User.objects.count() == 1
+
+
+@pytest.mark.django_db
+def test_register_with_password_with_no_letters(client: Client, user, error_messages):
+    resp = client.post(reverse('register'), data={'first_name': "Test", 'last_name': 'User', 'email': "test_@test.com",
+                                                  'password': "1234567?", 'agree_to_terms': True})
+    assert list(resp.context['form'].errors.values()) == [[error_messages['password_error']]]
+    assert User.objects.count() == 1
+
+
+@pytest.mark.django_db
+def test_register_with_password_with_no_numbers(client: Client, user, error_messages):
+    resp = client.post(reverse('register'), data={'first_name': "Test", 'last_name': 'User', 'email': "test_@test.com",
+                                                  'password': "test????", 'agree_to_terms': True})
+    assert list(resp.context['form'].errors.values()) == [[error_messages['password_error']]]
+    assert User.objects.count() == 1
+
+
+@pytest.mark.django_db
+def test_register_with_password_with_no_spec_symbols(client: Client, user, error_messages):
+    resp = client.post(reverse('register'), data={'first_name': "Test", 'last_name': 'User', 'email': "test_@test.com",
+                                                  'password': "test1234", 'agree_to_terms': True})
+    assert list(resp.context['form'].errors.values()) == [[error_messages['password_error']]]
+    assert User.objects.count() == 1
```

Comment:
There is no point using expensive high level tests, just for testing password validation. Test password validation function directly. Or better, use default Django password validation system.

https://docs.djangoproject.com/en/3.2/topics/auth/passwords/#module-django.contrib.auth.password_validation